test/deploy.py

```diff
@@ -0,0 +1,486 @@
+#!/usr/bin/python3
+import os
+import sys
+import pwd
+import json
+import yaml
+import time
+import pylxd
+import string
+import random
+import subprocess
+
+class Cleaner:
+    def __init__(self, model_file_path:string) -> None:
+        if os.path.exists(model_file_path):
+            # init client
+            self.client = pylxd.Client()
+            self.clean(model_file_path)
+        else:
+            print("Model File {} does not exists.".format(model_file_path))
+
+    def clean(self, model_file_path:string) -> None:
+        '''Clean LXC objects mentioned in the model file.'''
+        with open(model_file_path, 'r') as model_file:
+            model = json.loads(model_file.read())
+
+        vm_name = model['vm_name']
+        if self.client.instances.exists(vm_name):
+            print("Deleting VM {}".format(vm_name))
+            vm = self.client.virtual_machines.get(vm_name)
+            vm.stop(wait=True)
+            vm.delete(wait=True)
+
+        profile_name = model['profile']
+        if self.client.profiles.exists(profile_name):
+            print("Deleting VM Profile {}".format(vm_name))
+            self.client.profiles.get(profile_name).delete()
+
+        pool_name = model['storage_pool']
+        if self.client.storage_pools.exists():
+            pool = self.client.storage_pools.get(pool_name)
+
+            volumes = model['volumes']
+            for volume in volumes:
+                # Delete Volume.
+                print("Deleting Volume {} from Pool {}".format(volume, pool_name))
+                pool.volumes.get('custom', volume).delete()
+
+            # Delete Storage pool.
+            print("Deleting Pool {}".format(pool_name))
+            pool.delete()
+
+class DeployRunner:
+    # LXD Vars
+    test_tag = "oci-testing"
+    complete_repo_path = '/home/ubuntu'
+    cwd = os. getcwd()
+    usr = pwd.getpwuid(os.getuid())[0]
+    modelFilePath = ""
+    model = dict()
+
+    def __init__(self) -> None:
+        # Check LXD installed on host.
+        self.checkSnapsInstalled()
+        # Check if current user is part of the lxd user group.
+        self.checkUserInGroup()
+        # init client
+        self.client = pylxd.Client()
+        # Check if LXD is initialised.
+        self.checkLXDInitialised()
+        # File to store LXD virtual resource references.
+        self.modelFilePath = self.cwd+'/model-{}.json'.format(self._get_random_string(4))
+
+    def saveModelJson(self):
+        # Save model file.
```

Comment:
Possibly turn the comment into a docstring

Reply:
ack